Build failure: scaleway-cli

### Nixpkgs version

- Stable (26.05)

### Steps to reproduce

`nix build github:NixOS/nixpkgs?rev=ec942ba042dad5ef097e2ef3a3effc034241f011#scaleway-cli`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/job/nixos/release-26.05/nixpkgs.scaleway-cli.x86_64-linux

### Relevant log output

```console
error: builder for '/nix/store/38gi7y85kc93i5jb1xbfdlzw9i5hada2-scaleway-cli-2.54.0.drv' failed with exit code 1;
       last 25 log lines:
       > buildPhase completed in 2 minutes 18 seconds
       > Running phase: checkPhase
       > ok    github.com/scaleway/scaleway-cli/v2/cmd/scw     13.060s
       > ?       github.com/scaleway/scaleway-cli/v2/commands    [no test files]
       > ok      github.com/scaleway/scaleway-cli/v2/core        0.046s
       > --- FAIL: TestMarshal (0.00s)
       >     --- FAIL: TestMarshal/structWithMapsInSection (0.00s)
       >         marshal_test.go:155:
       >                     Error Trace:    /build/source/core/human/marshal_test.go:91
       >                     Error:          Not equal:
       >                              expected: "String              This is a string\nInt                 42\nBool                true\nStrings.0           s1\nStrings.1           s2\nTime                35 years ago\nStruct.String       -\nStruct.Int          0\nStruct.Bool         false\nStruct.Time         a long while ago\nStruct.Stringer     a stringer\nStructs.0.String    Nested string\nStructs.0.Int       0\nStructs.0.Bool      false\nStructs.0.Time      a long while ago\nStructs.0.Stringer  a stringer\nStringer            a stringer\nStringerPtr         a stringer\nSize                13 kB\nBytes               AAE=\n\nMap String List:\nkey1  v1 v2\nkey2  v3 v4\n\nMap:\nkey1  v1\nkey2  v2"
       >                                   actual  : "String              This is a string\nInt                 42\nBool                true\nStrings.0           s1\nStrings.1           s2\nTime                36 years ago\nStruct.String       -\nStruct.Int          0\nStruct.Bool         false\nStruct.Time         a long while ago\nStruct.Stringer     a stringer\nStructs.0.String    Nested string\nStructs.0.Int       0\nStructs.0.Bool      false\nStructs.0.Time      a long while ago\nStructs.0.Stringer  a stringer\nStringer            a stringer\nStringerPtr         a stringer\nSize                13 kB\nBytes               AAE=\n\nMap String List:\nkey1  v1 v2\nkey2  v3 v4\n\nMap:\nkey1  v1\nkey2  v2"
       >
       >                                   Diff:
       >                                   --- Expected
       >                                    +++ Actual
       >                              @@ -5,3 +5,3 @@
       >                                  Strings.1           s2
       >                                 -Time                35 years ago
       >                               +Time                36 years ago
       >                                Struct.String       -
       >                  Test:           TestMarshal/structWithMapsInSection
       > FAIL
       > FAIL   github.com/scaleway/scaleway-cli/v2/core/human  0.009s
       > FAIL
       For full logs, run:
         nix log /nix/store/38gi7y85kc93i5jb1xbfdlzw9i5hada2-scaleway-cli-2.54.0.drv
error: 1 dependencies of derivation '/nix/store/h2cpp7363l57vyd8jqhh17g52qad2w37-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/i55rjz25rdxhwr044w3j6bvcl9nbsqpw-home-manager-generation.drv' failed to build
```

### Additional context

Looks like this upstream PR would fix it: https://github.com/scaleway/scaleway-cli/pull/5567/
Am I allowed to do an "old man shouts at clouds" about test results that depend on the time the test is run? :(

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.0-94-generic, Ubuntu, 24.04.4 LTS (Noble Numbat), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/3p306srz83h9z9v0ma9xcxb8y8cdxkxj-source`


### Notify maintainers

@NickHu @techknowlogick @kashw2

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.
- [x] I assert that I have read the [automation/AI policy](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy) and that this issue report complies with it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/sc/scaleway-cli/package.nix
+++ b/pkgs/by-name/sc/scaleway-cli/package.nix
@@ -43,6 +43,16 @@ buildGo126Module (finalAttrs: {
   nativeInstallCheckInputs = [ versionCheckHook ];
   nativeCheckInputs = [ writableTmpDirAsHomeHook ];
 
+  checkFlags = [
+    # This subtest hardcodes a go-humanize relative-time string ("35 years ago")
+    # for a fixed 1990 date instead of computing it, so it breaks once enough
+    # wall-clock time passes (now "36 years ago"). go-humanize truncates the
+    # elapsed time by a fixed 360-day year (Year = 12*Month, Month = 30*Day), so
+    # diff/Year rolled 35 -> 36 on 2026-05-12. Upstream keeps bumping the
+    # literal by hand rather than fixing it, so skip the time-dependent subtest.
+    "-skip=^TestMarshal/structWithMapsInSection$"
+  ];
+
   postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
     $out/bin/scw autocomplete script basename=scw shell=bash >scw.bash
     $out/bin/scw autocomplete script basename=scw shell=fish >scw.fish

